astropy/utils/tests/test_misc.py

```diff
@@ -54,13 +54,24 @@ def test_skip_hidden():
 
 
 def test_JsonCustomEncoder():
+    from ... import units as u
     assert json.dumps(np.arange(3), cls=misc.JsonCustomEncoder) == '[0, 1, 2]'
     assert json.dumps(1+2j, cls=misc.JsonCustomEncoder) == '[1.0, 2.0]'
     assert json.dumps(set([1, 2, 1]), cls=misc.JsonCustomEncoder) == '[1, 2]'
     assert json.dumps(b'hello world \xc3\x85',
                       cls=misc.JsonCustomEncoder) == '"hello world \\u00c5"'
     assert json.dumps({1: 2},
                       cls=misc.JsonCustomEncoder) == '{"1": 2}'  # default
+    assert json.dumps({1: u.m}, cls=misc.JsonCustomEncoder) == '{"1": "m"}'
+    # Quantities
+    tmp = json.dumps({'a': 5*u.cm}, cls=misc.JsonCustomEncoder)
+    assert '"unit": "cm"' in tmp # Order of dict keys is unknown here
```

Comment:
Nitpicking: Two space between code and start of comment (PEP 8). You might want to use pycodestyle or pyflake or something similar to ensure PEP8 adherence. Not critical but would be nice, especially if someone decides to make code style check stricter on Travis CI.